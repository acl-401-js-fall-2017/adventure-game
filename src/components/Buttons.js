import React, { Component } from 'react';
import PropTypes from 'prop-types';


class Buttons extends Component {
  render(){
    const arrSort = ['cheese', 'crust', 'sauce'];
    const { floor, addToPizza, pickUp, pickUpVal, drop, pizzaIngredients, makePizza } = this.props;
    const check =  JSON.stringify(pizzaIngredients.sort()) === JSON.stringify(arrSort);
    const pizzaButton = check ? 
      <button key="7" className="make-pizza six" onClick={() => makePizza()}>bake pizza</button> :
      <button key="7" className="make-pizza six" onClick={() => addToPizza()}>Add to Pizza</button>;
    return [
      <label key="0" className="label-two">Pick up an item to take home</label>,
      <br key="1"/>,
      <button key="2" className="pick-up two" value={pickUpVal} onClick={({ target }) => pickUp(target.value)}>Pick Up</button>,
      <label key="3" className="label-three">Get rid of this item</label>,
      <br key=""/>,
      <button key="4" className="drop three" onClick= {() => drop()}>Drop</button>,
      <br key="5"/>,
      <label key="6" className="label-six">You have enough ingredients</label>,
      <div>
        {floor.key === '3rd Floor' && pizzaButton}
      </div>
    ];
  }
}
Buttons.propTypes = {
  addToPizza: PropTypes.func,
  pickUp: PropTypes.func,
  pickUpVal: PropTypes.string,
  drop: PropTypes.func,
  holding: PropTypes.string
};

export default Buttons;
  