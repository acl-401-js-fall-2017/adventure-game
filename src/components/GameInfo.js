import React, { Component } from 'react';
import PropTypes from 'prop-types';

class GameInfo extends Component {
  render() {
    const { holding, pizzaIngredients } = this.props;
    return (
      <div>
        <label> </label>
        <p>Holding: {holding}</p>
        <p>Pizza: {pizzaIngredients.join()}</p>
      </div>
    );
  }
}
GameInfo.propTypes = {
  pizzaIngredients: PropTypes.array,
  holding: PropTypes.string
};


export default GameInfo;