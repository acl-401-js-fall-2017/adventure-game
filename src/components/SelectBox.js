import React, { Component } from 'react';
import PropTypes from 'prop-types';


class SelectBox extends Component {
  render(){
    const { selectChange, options, className, name } = this.props;
    const Options = options.map((option, index) => {
      const optionElement = option.name === '3rd Floor' ? 
        <option key={index} value={index} default>Home</option> :
        <option key ={index} value={index}>{option.name || option}</option>;
      return optionElement;
    });
    return(
      <select className={className} name={name} onChange={({ target }) => selectChange(target.value)}>
        { Options }
      </select>
    );
  }
}
SelectBox.PropTypes = {
  selectChange: PropTypes.func,
  options: PropTypes.array,
  className: PropTypes.string,
  string: PropTypes.string
};

export default SelectBox;
  