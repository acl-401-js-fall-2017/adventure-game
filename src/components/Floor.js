import React, { Component } from 'react';
import GameInfo from './GameInfo';
import PropTypes from 'prop-types';


class Floor extends Component {
  render() {
    const { floorScript, holding, pizzaIngredients } = this.props;
    return (
      <div>
        <header className="App-header">
          <h1 className="flash">Shane is starving at home</h1>
          <p className="msg">{floorScript}</p>
          <GameInfo holding={holding} pizzaIngredients={pizzaIngredients} />
          <img className="pizza" alt="pizza" src="./images/pizza2.png" />
        </header>
      </div>
    );
  }
}
Floor.propTypes = {
  floorScript: PropTypes.string
};

export default Floor;