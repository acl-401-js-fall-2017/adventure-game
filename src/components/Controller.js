import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Buttons from './Buttons';
import SelectBox from './SelectBox';

class Controller extends Component{
  render(){  
    const { floor, floors, pizzaIngredients, floorChange, pickUpVal, pickUp, addToPizza, drop, handlePickUpValue, makePizza } = this.props;
    return(
      <div className="wrapper">
        <label className="label-one">The Elevator</label>
        <SelectBox options={floors} className="dropDown one" name="elevator"
          selectChange={floorNumber => floorChange(floorNumber)} />
        <Buttons pickUpVal={pickUpVal} pickUp={item => pickUp(item)} pizzaIngredients={pizzaIngredients}
          addToPizza={item => addToPizza(item)} drop={() => drop()} floor={floor} makePizza={makePizza}/>
        <img id="pics" alt={floor.alt} src={floor.img} />
        <label className="label-five">Items in the Room</label>
        <SelectBox options={floor.items} className="dropDown five" name="items-in-room" 
          selectChange={item => handlePickUpValue(item)}/>
      </div>
    );
  }
}
Controller.propTypes = {
  floor: PropTypes.object,
  floors: PropTypes.array,
  floorChange: PropTypes.func,
  addToPizza: PropTypes.func,
  pickUp: PropTypes.func,
  pickUpVal: PropTypes.string,
  drop: PropTypes.func,
  handlePickUpValue: PropTypes.func,
  holding: PropTypes.string
};
  
export default Controller;