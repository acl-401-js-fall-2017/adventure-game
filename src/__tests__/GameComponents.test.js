import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import GameInfo from '../components/GameInfo';
import SelectBox from '../components/SelectBox';
import floors from '../Floors';
import Buttons from '../components/Buttons';
import Controller from '../components/Buttons';
import Floor from '../components/Floor';


describe('App', () => {

  it('should render game info component', () => {
    const wrapper = shallow(<GameInfo holding="sauce" pizzaIngredients={['cheese', 'crust']} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render SelectBox component', () => {
    const wrapper = shallow(<SelectBox options={floors} className="dropDown one" name="elevator"
      selectChange={floorNumber => floorNumber} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render Buttons component', () => {
    const wrapper = shallow(<Buttons pickUpVal="cheese" pizzaIngredients={['sauce', 'crust']} floor={floors[0]}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render Controller component', () => {
    const wrapper = shallow(<Controller floor={floors[0]} floors={floors}  pizzaIngredients={['sauce', 'crust']}
      pickUpVal="sauce" holding="crust"/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render Floor component', () => {
    const wrapper = shallow(<Floor floorScript ={floors[1].message} holding="sauce" pizzaIngredients={['sauce', 'crust']}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });


});