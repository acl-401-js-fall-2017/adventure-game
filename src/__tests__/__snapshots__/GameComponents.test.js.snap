// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render Buttons component 1`] = `
Array [
  <label
    className="label-two"
    key="0"
/>,
  <br
    key="1"
/>,
  <button
    className="pick-up two"
    key="2"
    onClick={[Function]}
    value="cheese"
/>,
  <label
    className="label-three"
    key="3"
/>,
  <br
    key=""
/>,
  <button
    className="drop three"
    key="4"
    onClick={[Function]}
/>,
  <br
    key="5"
/>,
  <label
    className="label-six"
    key="6"
/>,
  <div />,
]
`;

exports[`App should render Controller component 1`] = `
Array [
  <label
    className="label-two"
    key="0"
/>,
  <br
    key="1"
/>,
  <button
    className="pick-up two"
    key="2"
    onClick={[Function]}
    value="sauce"
/>,
  <label
    className="label-three"
    key="3"
/>,
  <br
    key=""
/>,
  <button
    className="drop three"
    key="4"
    onClick={[Function]}
/>,
  <br
    key="5"
/>,
  <label
    className="label-six"
    key="6"
/>,
  <div />,
]
`;

exports[`App should render Floor component 1`] = `
<div>
  <header
    className="App-header"
  >
    <h1
      className="flash"
    >
      Shane is starving at home
    </h1>
    <p
      className="msg"
    >
      You inadvertently knock on the door of your landlord when you are late with the rent. You apologize profusely but the landlord has no mercy. He takes all of the pizza ingredients that you have accumulated so far from your pantry.
    </p>
    <GameInfo
      holding="sauce"
      pizzaIngredients={
        Array [
          "sauce",
          "crust",
        ]
      }
    />
    <img
      alt="pizza"
      className="pizza"
      src="./images/pizza2.png"
    />
  </header>
</div>
`;

exports[`App should render SelectBox component 1`] = `
<select
  className="dropDown one"
  name="elevator"
  onChange={[Function]}
>
  <option
    key="0"
    value={0}
  >
    Ground Floor
  </option>
  <option
    key="1"
    value={1}
  >
    1st Floor
  </option>
  <option
    key="2"
    value={2}
  >
    2nd Floor
  </option>
  <option
    default={true}
    key="3"
    value={3}
  >
    Home
  </option>
  <option
    key="4"
    value={4}
  >
    4th Floor
  </option>
  <option
    key="5"
    value={5}
  >
    5th Floor
  </option>
  <option
    key="6"
    value={6}
  >
    6th Floor
  </option>
</select>
`;

exports[`App should render game info component 1`] = `
<div>
  <label>
     
  </label>
  <p>
    Holding: 
    sauce
  </p>
  <p>
    Pizza: 
    cheese,crust
  </p>
</div>
`;
